### PR TITLE
[kube-prometheus-stack] Remove app label on ruleSelector

### PR DESCRIPTION
**Pull request by nlamirault**
#### What this PR does / why we need it:

ruleSelector does not have same `matchLabels` field of the others selectors.

```yaml
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
...
spec:
  podMonitorSelector:
    matchLabels:
      release: kube-prometheus-stack
  ...
  probeSelector:
    matchLabels:
      release: kube-prometheus-stack
   ...
  ruleSelector:
    matchLabels:
      app: kube-prometheus-stack
      release: kube-prometheus-stack
   ...
  serviceMonitorSelector:
    matchLabels:
      release: kube-prometheus-stack
```

This PR remove the `app` label. Like that all selectors are identical.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — COMMENTED** [review]
Could you please add explanation why its removed and why the you've raised the minor version?

**Comment by nlamirault** [comment]
@monotek Sorry i think i forget to save my previous comment :(
Due to matchLabels change i update the minor version. Should a patch be better?

**Review by monotek — CHANGES_REQUESTED** [review]
Yes, if its a bugfix and no new features added, thje patch version should be raised.

**Comment by gkarthiks** [comment]
> Could you please add explanation why its removed and why the you've raised the minor version?

hi @nlamirault as requested by @monotek, can you please add an explanation for why its being removed clearly?

**Review by monotek — APPROVED** [review]

**Comment by nlamirault** [comment]
> > Could you please add explanation why its removed and why the you've raised the minor version?
> 
> hi @nlamirault as requested by @monotek, can you please add an explanation for why its being removed clearly?

@gkarthiks it's just to make all selectors work the same

**Review by gkarthiks — APPROVED** [review]

File: charts/kube-prometheus-stack/Chart.yaml [hunk 1]
```diff
@@ -17,8 +17,8 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 23.2.0
-appVersion: 0.52.1
+version: 23.3.0
+appVersion: 0.52.0
```
Comment (mrueg) [inline charts/kube-prometheus-stack/Chart.yaml:21]: You downgraded the prometheus operator version here.

@nlamirault @gkarthiks

**Pull request merged**